field.go

```diff
@@ -300,10 +300,12 @@ func (f *Field) saveAvailableShards() error {
 }
 
 func (f *Field) unprotectedSaveAvailableShards() error {
-	// Open or create file.
 	path := filepath.Join(f.path, ".available.shards")
+	// Create a temporary file to snapshot to.
+	snapshotPath := path + snapshotExt
 
-	file, err := os.OpenFile(path, os.O_WRONLY|os.O_CREATE|os.O_TRUNC, 0666)
+	// Open or create file.
+	file, err := os.OpenFile(snapshotPath, os.O_WRONLY|os.O_CREATE|os.O_TRUNC, 0666)
 	if err != nil {
 		return errors.Wrap(err, "opening available shards file")
```

Comment:
let's update this error message to reflect that it is the temp file we're opening